pkgs/by-name/li/libguestfs-appliance-debian/package.nix

```diff
@@ -0,0 +1,113 @@
+{
+  lib,
+  stdenvNoCC,
+  vmTools,
+  fetchurl,
+  runCommand,
+  qemu-utils,
+}:
+let
+  debianSnapshot = "20250227T210548Z";
+
+  debianArch =
+    {
+      "x86_64-linux" = "amd64";
+      "aarch64-linux" = "arm64";
+    }
+    .${stdenvNoCC.hostPlatform.system};
+
+  debClosureForArch =
+    arch:
+    vmTools.debClosureGenerator rec {
+      name = "libguestfs-appliance-deb-closure-${arch}";
+      packages = vmTools.commonDebianPackages ++ [
+        "libguestfs-tools"
+        "linux-image-${arch}" # required by supermin
+      ];
+      packagesLists = [
+        (fetchurl {
+          url = "${urlPrefix}/dists/testing/main/binary-${arch}/Packages.xz";
+          hash =
+            {
+              "amd64" = "sha256-llPpp2k6wDAHTXFMT24ANUTIcj4SDsE16oc35la4JYk=";
+              "arm64" = "sha256-EU3tANE/SB9rM1IZw80GT5UyewpWjmfcb35HYd5cHPo=";
+            }
+            .${arch};
+        })
+      ];
+      urlPrefix = "https://snapshot.debian.org/archive/debian/${debianSnapshot}";
+    };
+
+  debClosures = runCommand "libguestfs-deb-closures" { } (
+    lib.concatMapStringsSep "\n"
+      (arch: "install -Dm444 ${debClosureForArch arch} $out/deb-closure-${arch}.nix")
+      [
+        "amd64"
+        "arm64"
+      ]
+  );
+
+  diskImage = vmTools.fillDiskWithDebs {
+    name = "debian-testing-${debianArch}";
+    fullName = "Debian testing (${debianArch})";
+    debs =
+      {
+        amd64 = import ./deb-closure-amd64.nix { inherit fetchurl; };
+        arm64 = import ./deb-closure-arm64.nix { inherit fetchurl; };
+      }
+      .${debianArch};
```

Comment:
This requires an or with a throw otherwise we get a syntax error on other arches

Reply:
`debianArch` is by definition one of these two.

Reply:
That's not syntax error. It's runtime error. This works perfectly fine in repl: `let x = "foo"; in if false then { a = 2; }.${x} else 3`.